RUFAS/routines/field/crop/nitrogen_incorporation.py

```diff
@@ -396,7 +397,19 @@ def determine_nutrient_shape_parameters(
             If half_mature_heat_fraction equals mature_heat_fraction.
 
         """
+        info_map = {
+            "class": NitrogenIncorporation.__class__.__name__,
+            "function": NitrogenIncorporation.determine_nutrient_shape_parameters.__name__,
+        }
+        om = OutputManager()
         if mature_heat_fraction == half_mature_heat_fraction:  # leads to divide by 0
+            om.add_error(
+                "Equal half_mature_heat_fraction and mature_heat_fraction.",
+                f"half_mature_heat_fraction must not equal mature_heat_fraction,"
+                f" received mature_heat_fraction is {mature_heat_fraction},"
+                f" half_mature_heat_fraction is {half_mature_heat_fraction}",
+                info_map,
```

Comment:
```suggestion
                "A crop's half mature heat fraction and mature heat fraction are equal.",
                f"Half mature heat fraction and mature heat fraction are both {mature_heat_fraction}, this results in a divide by zero error."
                info_map,
```
You don't have to use this exact suggestion, but keep in mind that it is good to write both the cause and effect in error and warning messages. Ideally, the info map would also contain information like the name of the crop and simulation date when this occurs. But that can't be done without changing the method args or making it non-static, neither of which would be worth while here IMO.